### PR TITLE
added version line to docker, added schema for prisma

### PR DESCRIPTION
**Pull request by billwang7599**
Docker: added 
```version: "3"```

Prisma: added schema. Works with ```npx prisma generate```

### CODE REVIEW

File: frontend/prisma/schema.prisma [hunk 1]
```diff
@@ -15,16 +15,46 @@ datasource db {
...
+  cid         String
+  clubs       Clubs          @relation(fields: [cid], references: [cid])
+  type        String
+  categories  Categories     @relation(fields: [type], references: [type])
+  @@unique([cid, type]) // Ensuring that each combination of cid and type is unique
+}
+
+model ClubInfo {
+  cid         String   @unique
+  clubs       Clubs    @relation(fields: [cid], references: [cid])
```
Comment (xusandrew) [inline frontend/prisma/schema.prisma]: can u rename this field to just "club"

File: frontend/prisma/schema.prisma [hunk 2]
```diff
@@ -15,16 +15,46 @@ datasource db {
...
-  createdAt DateTime @default(now())
-  updatedAt DateTime @updatedAt
-  email     String?  @unique
-  image     String?
-  role      Role     @default(USER)
+model Clubs {
+  cid         String         @id @default(uuid())
+  email       String         @unique
+  password    String
+  info        ClubInfo[]
```
Comment (xusandrew) [inline frontend/prisma/schema.prisma]: do we want a club to have multiple ClubInfo?

File: frontend/prisma/schema.prisma [hunk 3]
```diff
@@ -15,16 +15,46 @@ datasource db {
...
 }
 
-model User {
-  id        Int      @id @default(autoincrement())
-  createdAt DateTime @default(now())
-  updatedAt DateTime @updatedAt
-  email     String?  @unique
-  image     String?
-  role      Role     @default(USER)
+model Clubs {
```
Comment (xusandrew) [inline frontend/prisma/schema.prisma]: can you rename all the models such that they are not plural, so Clubs -> Club, Events -> Event, makes things less confusing

File: frontend/prisma/schema.prisma [hunk 4]
```diff
@@ -15,16 +15,46 @@ datasource db {
...
+  description String?
+  location    String
+  start_time  DateTime
+  end_time    DateTime
+  posted_time DateTime @default(now())
 }
+
+model Categories {
+  type        String         @id
+  clubs       ClubCategories[]
```
Comment (xusandrew) [inline frontend/prisma/schema.prisma]: can u rename this field to just "club"

File: frontend/prisma/schema.prisma [hunk 5]
```diff
@@ -15,16 +15,46 @@ datasource db {
...
+  categories  ClubCategories[]
 }
 
-enum Role {
-  USER
-  ADMIN
+model Events {
+  eid         String   @id @default(uuid())
+  cid         String
+  clubs       Clubs    @relation(fields: [cid], references: [cid])
```
Comment (xusandrew) [inline frontend/prisma/schema.prisma]: can u rename this field to just "club"

File: frontend/prisma/schema.prisma [hunk 6]
```diff
@@ -15,16 +15,46 @@ datasource db {
...
+  type        String         @id
+  clubs       ClubCategories[]
+}
+
+model ClubCategories {
+  cid         String
+  clubs       Clubs          @relation(fields: [cid], references: [cid])
+  type        String
+  categories  Categories     @relation(fields: [type], references: [type])
+  @@unique([cid, type]) // Ensuring that each combination of cid and type is unique
```
Comment (xusandrew) [inline frontend/prisma/schema.prisma]: beauty

File: docker-compose.yml [hunk 7]
```diff
@@ -13,4 +14,4 @@ services:
       - POSTGRES_DB=postgres
 volumes:
   my-db:
-    driver: local
\ No newline at end of file
+    driver: local
```
Comment (xusandrew) [inline docker-compose.yml:17]: swag

**Review by xusandrew — COMMENTED** [review]
@billwang7599 ur so hot

can u make a migration for this as well, `npx prisma migrate dev` and include that in this pr

File: frontend/prisma/schema.prisma [hunk 8]
```diff
@@ -15,16 +15,46 @@ datasource db {
...
-  updatedAt DateTime @updatedAt
-  email     String?  @unique
-  image     String?
-  role      Role     @default(USER)
+model Clubs {
+  cid         String         @id @default(uuid())
+  email       String         @unique
+  password    String
+  info        ClubInfo[]
+  events      Events[]
```
Comment (mxc-maggiechen) [inline frontend/prisma/schema.prisma]: just wondering, how would we be using this column? i was thinking we would be using the events table whenever we want to access events

File: frontend/prisma/schema.prisma [hunk 9]
```diff
@@ -15,16 +15,46 @@ datasource db {
...
-enum Role {
-  USER
-  ADMIN
+model Events {
+  eid         String   @id @default(uuid())
+  cid         String
+  clubs       Clubs    @relation(fields: [cid], references: [cid])
+  title       String
+  description String?
+  location    String
```
Comment (mxc-maggiechen) [inline frontend/prisma/schema.prisma]: would it be better to have location as optional?

File: frontend/prisma/schema.prisma (hunk 4 above)
Comment (mxc-maggiechen) [reply frontend/prisma/schema.prisma]: also wondering how would we be using this field

File: frontend/prisma/schema.prisma [hunk 10]
```diff
@@ -15,16 +15,46 @@ datasource db {
...
+  posted_time DateTime @default(now())
 }
+
+model Categories {
+  type        String         @id
+  clubs       ClubCategories[]
+}
+
+model ClubCategories {
+  cid         String
```
Comment (mxc-maggiechen) [inline frontend/prisma/schema.prisma]: would it be good if we have the id + uuid restriction thing for this?

File: frontend/prisma/schema.prisma [hunk 11]
```diff
@@ -15,16 +15,46 @@ datasource db {
...
+  clubs       Clubs          @relation(fields: [cid], references: [cid])
+  type        String
+  categories  Categories     @relation(fields: [type], references: [type])
+  @@unique([cid, type]) // Ensuring that each combination of cid and type is unique
+}
+
+model ClubInfo {
+  cid         String   @unique
```
Comment (mxc-maggiechen) [inline frontend/prisma/schema.prisma]: same here, would it be good if we have the @id uuid thing here

File: frontend/prisma/schema.prisma (hunk 2 above)
Comment (billwang7599) [reply frontend/prisma/schema.prisma]: No, I don't think so

Comment (billwang7599) [reply frontend/prisma/schema.prisma]: Do we want to require a club to have club info?

File: frontend/prisma/schema.prisma (hunk 8 above)
Comment (billwang7599) [reply frontend/prisma/schema.prisma]: It's just to establish a link between Events and Clubs. This is essentially saying the events table has a many to one relationship to clubs

File: frontend/prisma/schema.prisma (hunk 9 above)
Comment (billwang7599) [reply frontend/prisma/schema.prisma]: Sure, but what event would have an optional location? Even online events can be like discord

File: frontend/prisma/schema.prisma (hunk 10 above)
Comment (billwang7599) [reply frontend/prisma/schema.prisma]: Like the @@unique thing below?

File: frontend/prisma/migrations/20240618055129_sample_init/migration.sql
Comment (xusandrew) [inline frontend/prisma/migrations/20240618055129_sample_init/migration.sql:1]: looks good

File: frontend/prisma/schema.prisma [hunk 12]
```diff
@@ -12,19 +12,49 @@ generator client {
 
 datasource db {
   provider = "postgresql"
-  url      = env("DATABASE_URL")
+  url      = "postgresql://postgres:postgres@localhost:5555/DATABASE"
```
Comment (xusandrew) [inline frontend/prisma/schema.prisma]: @billwang7599
myb forgot to add this into the readme, but you should copy the `.env.example` file and name it `.env`
that way it'll recognize the "DATABASE_URL" environment variable

if that dosent work lmk

File: frontend/prisma/schema.prisma [hunk 13]
```diff
@@ -12,19 +12,49 @@ generator client {
...
-  email     String?  @unique
-  image     String?
-  role      Role     @default(USER)
+model Club {
+  cid      String         @id @default(uuid())
+  email    String         @unique
+  password String
+  event    Event[] // Event referencing Club, one-to-many relation
+  category ClubCategory[] // ClubCategory, many-to-many relation
+  ClubInfo ClubInfo? // Defines a one-to-one relationship, optional (idk why)
```
Comment (xusandrew) [inline frontend/prisma/schema.prisma]: should be required im pretty sure

Comment (billwang7599) [reply frontend/prisma/schema.prisma]: a club can have 0 or 1 profile essentially. if we want it to be required, we may as well just include it in the table itself

**Review by xusandrew — APPROVED** [review]

**Pull request merged**